### PR TITLE
[feature/ASV-1314] Extract aptoide theme

### PR DESCRIPTION
**Pull request by JDSM01**
**What does this PR do?**

   This ticket makes sure that all views are using the theme choose in gradle instead of colors or drawables;
  Some of the old orange colors were changed on this ticket to the new ones
  I tried to check all the flows, but there’s probably something that I missed (hope not)
 Some drawables that were not being used were deleted in the process

**Database changed?**

   No

**Where should the reviewer start?**

themes.xml

**How should this be manually tested?**

  Test as many flows as possible with vanilla and cobrand version;
  Use a color different than red or green (those were the one which I tested) (go to gradle properties and change the value of cobrand-theme);

**What are the relevant tickets?**

  https://aptoide.atlassian.net/browse/ASV-1314

**Questions:**

**Code Review Checklist**

- [ ] Architecture
- [ ] Documentation on public interfaces
- [ ] Database changed?
- [ ] If yes - Database migration?
- [ ] Remove comments & unused code & forgotten testing Logs
- [ ] Codestyle
- [ ] Partners build
- [ ] Unit tests pass
- [ ] Functional QA tests pass

### CODE REVIEW

File: app/src/main/java/cm/aptoide/pt/app/view/AppViewPresenter.java [hunk 1]
```diff
@@ -854,11 +854,10 @@ private void handleInstallButtonClick() {
   }
 
   private void showRecommendsDialog(boolean isLoggedIn, String packageName) {
+    view.showRecommendsDialog();
```
Comment (paquiquinho) [inline app/src/main/java/cm/aptoide/pt/app/view/AppViewPresenter.java]: Why are we changing this here? Won't we always show the dialog as if the user is logged in regardless of the logic?

File: app/src/main/java/cm/aptoide/pt/home/BottomNavigationActivity.java [hunk 2]
```diff
@@ -30,6 +32,7 @@
   private Animation animationdown;
 
   @Override protected void onCreate(Bundle savedInstanceState) {
+    ThemeUtils.setAptoideTheme(this, BuildConfig.APTOIDE_THEME);
```
Comment (paquiquinho) [inline app/src/main/java/cm/aptoide/pt/home/BottomNavigationActivity.java]: Why are we going to buildConfig here instead of injecting?

File: app/src/main/java/cm/aptoide/pt/link/CustomTabsHelper.java [hunk 3]
```diff
@@ -92,8 +94,10 @@ public void openInChromeCustomTab(String url, Context context) {
         new Intent(context.getApplicationContext(), CustomTabNativeReceiver.class);
     PendingIntent pendingIntent =
         PendingIntent.getBroadcast(context.getApplicationContext(), 0, openInNativeIntent, 0);
+    int toolbarColor = StoreTheme.get(BuildConfig.APTOIDE_THEME)
```
Comment (paquiquinho) [inline app/src/main/java/cm/aptoide/pt/link/CustomTabsHelper.java]: Is there no way to inject the theme here as well?

File: app/src/main/java/cm/aptoide/pt/ApplicationComponent.java [hunk 4]
```diff
@@ -26,4 +27,6 @@ ActivityComponent plus(ActivityModule activityModule,
   void inject(InstalledIntentService installedIntentService);
 
   void inject(PullingContentService pullingContentService);
+
+  void inject(DescriptionFragment descriptionFragment);
```
Comment (paquiquinho) [inline app/src/main/java/cm/aptoide/pt/ApplicationComponent.java]: This should be (and is being) injected in fragmentComponent

File: app/src/main/res/drawable/oval_white.xml [hunk 5]
```diff
@@ -1,5 +1,5 @@
 <?xml version="1.0" encoding="utf-8"?>
 <shape xmlns:android="http://schemas.android.com/apk/res/android"
     android:shape="oval">
-  <solid android:color="@color/aptoide_orange"/>
+  <solid android:color="@color/default_orange_gradient_end"/>
```
Comment (paquiquinho) [inline app/src/main/res/drawable/oval_white.xml]: This class in v21 has color "colorPrimaryDark". Why are we using our default color here?

File: app/src/main/java/cm/aptoide/pt/view/fragment/DescriptionFragment.java [hunk 6]
```diff
@@ -98,16 +96,9 @@ public static String getSTORE_THEME() {
     return DescriptionFragment.STORE_THEME;
   }
 
-  public static String getDESCRIPTION() {
-    return DescriptionFragment.DESCRIPTION;
-  }
-
-  public static String getAPP_NAME() {
```
Comment (paquiquinho) [inline app/src/main/java/cm/aptoide/pt/view/fragment/DescriptionFragment.java]: why were these removed?

File: app/src/main/res/values/styles_aptoide_text_view.xml [hunk 7]
```diff
@@ -224,4 +223,72 @@
     <item name="android:textColor">@color/youtube</item>
   </style>
 
+  <!-- Theme Color -->
+
+  <style name="Aptoide.TextView.Regular.L.Theme">
+    <item name="android:textSize">19sp</item>
```
Comment (paquiquinho) [inline app/src/main/res/values/styles_aptoide_text_view.xml]: You don't need the text sizes here, they are inherited from "Aptoide.TextView.Regular.L". Look at "Aptoide.TextView.Medium.M.BlackAlpha" for reference

File: app/src/main/java/cm/aptoide/pt/view/BaseActivity.java [hunk 8]
```diff
@@ -15,6 +15,8 @@
 
   @Override protected void onCreate(@Nullable Bundle savedInstanceState) {
     super.onCreate(savedInstanceState);
```
Comment (paquiquinho) [inline app/src/main/java/cm/aptoide/pt/view/BaseActivity.java]: not sure if it is possible to inject stuff here bt it'd be worth the try

File: app/src/main/res/drawable/oval_white.xml (hunk 5 above)
Comment (JDSM01) [reply app/src/main/res/drawable/oval_white.xml]: It’s not possible to use ?attr in drawable for api below 21, we could create a drawable for each theme color, but given the places where the resource is being used I would just leave it as it is

File: app/src/main/java/cm/aptoide/pt/ApplicationComponent.java (hunk 4 above)
Comment (JDSM01) [reply app/src/main/java/cm/aptoide/pt/ApplicationComponent.java]: Done

File: app/src/main/res/values/styles_aptoide_text_view.xml (hunk 7 above)
Comment (JDSM01) [reply app/src/main/res/values/styles_aptoide_text_view.xml]: Done

File: app/src/main/java/cm/aptoide/pt/view/BaseActivity.java (hunk 8 above)
Comment (JDSM01) [reply app/src/main/java/cm/aptoide/pt/view/BaseActivity.java]: It is. Done
